controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -251,7 +251,17 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, cluster *
 		return r.upgradeControlPlane(ctx, cluster, kcp, ownedMachines, requireUpgrade)
 	}
 
-	// If we've made it this far, we we can assume that all ownedMachines are up to date
+	//If we've made it this far, we can assume that all control-plane machines are up to date
```

Comment:
This isn't true if we're about to create the first replica, is it? See line 259/269 below.